File: admin/src/main/frontend/app/scripts/app/service/GroupService.ts
'use strict';

module cda.app.service {

  import Group = cda.app.model.Group;

  export interface IGroupService {
    getGroups(): ng.IPromise<Group[]>;
    getGroup(groupId: number): ng.IPromise<Group>;
    saveGroup(group: Group): ng.IPromise<Group>;
  }

  class GroupService implements IGroupService {
    static $inject = ['Restangular', 'locationParams', '$window'];

    constructor(private restangular: restangular.IService, private locationParams, private $window: ng.IWindowService) {
    }

    getGroups(): ng.IPromise<Group[]> {
      return this.restangular.all('group').getList();
    }

    getGroup(groupId: number): ng.IPromise<Group> {
      return this.restangular.one('group', groupId).get();
    }

    saveGroup(group: Group): ng.IPromise<Group> {
      return this.restangular.all('group').post(group);
    }
  }

  angular.module("adminApp").service("GroupService", GroupService);
}

File: admin/src/main/java/ru/cg/cda/admin/service/GroupService.java
package ru.cg.cda.admin.service;

import java.util.List;

import ru.cg.cda.admin.dto.GroupDTO;
import ru.cg.cda.database.bean.Group;


/**
 * @author Badamshin
 */
public interface GroupService {
  GroupDTO getGroup(Long groupId);

  List<GroupDTO> getGroups();

  GroupDTO saveGroup(GroupDTO groupDTO);

  GroupDTO convertGroup(Group group);

  List<GroupDTO> convertGroups(List<Group> groups);
}
